common/system/filtering/altitude_kalman.cpp

```diff
@@ -1,13 +1,23 @@
 #include "altitude_kalman.h"
 
+#include "hardware_manager.h"
+
 constexpr const double AltitudeKalman::DEFAULT_KALMAN_GAIN[];
 
-AltitudeKalman::AltitudeKalman(double dt) : m_dt(dt) {}
+AltitudeKalman::AltitudeKalman(double dt)
+    : m_dt(dt), firstTime(true), m_lastPredictTime(hm_millis()) {}
 
 void AltitudeKalman::predict(const double az) {
   // x_k+1 = A x_k + B u_k
   // x_k+1 = [pos] = [old pos + vel * m_dt] + [1/2 acceleration m_dt^2]
   //         [vel]   [old vel           ]   [acceleration * m_dt    ]
+
+  m_dt = (hm_getSensorData()->timestampMs - m_lastPredictTime) / 1000.0;
+  m_lastPredictTime = hm_getSensorData()->timestampMs;
+  if (firstTime) {
```

Comment:
I think this might be something we added to try to fix the sim timing issues, but we don't think that's necessarily an issue associated with the lockout smoothing right? Not necessarily a problem to include it if it works but it would be worth testing without this fix if you haven't done that.

Reply:
I don't have physical hardware so I'm not entirely sure how it behaves there. I can potentially see dt overflowing if we aren't careful. I will test in sim tho

Reply:
Oh I'm stupid I see what you mean. I rolled back these changes, is that better?

Reply:
Yeah that's what I had meant. If you've been able to test it I think it looks good.